ImperatorToCK3/CK3/World.cs

```diff
@@ -314,40 +363,95 @@ Imperator.World impWorld
 			}
 		}
 
-		private void OverWriteCountiesHistory(Date ck3BookmarkDate) {
+		private void OverWriteCountiesHistory(List<Governorship> governorships, Date ck3BookmarkDate) {
 			Logger.Info("Overwriting counties' history.");
 			foreach (var title in LandedTitles.Values) {
-				if (title.Rank == TitleRank.county && title.CapitalBaronyProvince > 0) { // title is a county and its capital province has a valid ID (0 is not a valid province in CK3)
-					if (!Provinces.ContainsKey(title.CapitalBaronyProvince)) {
-						Logger.Warn($"Capital barony province not found {title.CapitalBaronyProvince}");
-					} else {
-						var ck3CapitalBaronyProvince = Provinces[title.CapitalBaronyProvince];
-						var impProvince = ck3CapitalBaronyProvince.ImperatorProvince;
-						if (impProvince is not null) {
-							var impCountry = impProvince.OwnerCountry.Value;
-							if (impCountry is not null && impCountry.CountryType != CountryType.rebels) {
-								var ck3Country = impCountry.CK3Title;
-								if (ck3Country is not null) {
-									if (Characters.TryGetValue(ck3Country.GetHolderId(ck3BookmarkDate), out var holder)) {
-										title.ClearHolderHistory();
-										title.SetHolderId(holder.ID, ck3Country.GetDateOfLastHolderChange());
-									}
-									title.DeFactoLiege = null;
-									countyHoldersCache.Add(title.GetHolderId(ck3BookmarkDate));
-								}
-							} else { // e.g. uncolonised Imperator province
-								title.SetHolderId("0", ck3BookmarkDate);
-								title.DeFactoLiege = null;
+				if (title.Rank != TitleRank.county) {
+					continue;
+				}
+				if (title.CapitalBaronyProvince <= 0) {
+					// title's capital province has an invalid ID (0 is not a valid province in CK3)
+					continue;
+				}
+
+				if (!Provinces.ContainsKey(title.CapitalBaronyProvince)) {
+					Logger.Warn($"Capital barony province not found {title.CapitalBaronyProvince}");
+					continue;
+				}
+				var ck3CapitalBaronyProvince = Provinces[title.CapitalBaronyProvince];
+				var impProvince = ck3CapitalBaronyProvince.ImperatorProvince;
+
+				if (impProvince is not null) {
```

Comment:
Reverse the condition and make the null case continue the loop. Then you can move the rest out one indentation level.